tests/test_subscriptions.py

```diff
@@ -146,22 +149,25 @@ def test_subscription_registration_succeeds_when_query_does_not_match_mappings(s
         self.assertIn('uuid', resp_obj.json)
 
     def test_get(self):
-        find_uuid = self._put_subscription()
-
-        # Normal request
-        url = str(UrlBuilder()
-                  .set(path="/v1/subscriptions/" + str(find_uuid))
-                  .add_query("replica", self.replica.name)
-                  .add_query("subscription_type", "elasticsearch"))
-        resp_obj = self.assertGetResponse(
-            url,
-            requests.codes.okay,
-            headers=get_auth_header())
-        json_response = resp_obj.json
-        self.assertEqual(self.sample_percolate_query, json_response['es_query'])
-        self.assertEqual(self.endpoint, Endpoint.from_subscription(json_response))
-        self.assertEquals(self.hmac_key_id, json_response['hmac_key_id'])
-        self.assertNotIn('hmac_secret_key', json_response)
+        try:
+            find_uuid = self._put_subscription()
+
+            # Normal request
+            url = str(UrlBuilder()
+                      .set(path="/v1/subscriptions/" + str(find_uuid))
+                      .add_query("replica", self.replica.name)
+                      .add_query("subscription_type", "elasticsearch"))
+            resp_obj = self.assertGetResponse(
+                url,
+                requests.codes.okay,
+                headers=get_auth_header())
+            json_response = resp_obj.json
+            self.assertEqual(self.sample_percolate_query, json_response['es_query'])
+            self.assertEqual(self.endpoint, Endpoint.from_subscription(json_response))
+            self.assertEquals(self.hmac_key_id, json_response['hmac_key_id'])
+            self.assertNotIn('hmac_secret_key', json_response)
+        finally:
+            self._cleanup_subscription(json_response['uuid'])
```

Comment:
If it's `self._cleanup_subscription(find_uuid)`, then it will be cleaned up if the `PUT` succeeds and the `GET` fails.

Reply:
resolved, good catch, i guess it is possible for the assertEquals() check to fail here.....